kernel/sched.c

```diff
@@ -1594,6 +1594,12 @@ static int cpu_mask_mod(k_tid_t thread, uint32_t enable_mask, uint32_t disable_m
 {
 	int ret = 0;
 
+#if defined(CONFIG_ASSERT) && defined(CONFIG_SCHED_CPU_MASK_PIN_ONLY)
+	__ASSERT((thread->base.thread_state != _THREAD_DEAD) ||
+		 (thread->base.thread_state != _THREAD_PRESTART),
```

Comment:
While it's true that it's safe to call this on a dead thread, it's not useful.  Thread structs that get restarted get all their state reinitialized (thus losing this setting before it can have any effect) and enter PRESTARTED anyway.

Reply:
done, just removed the check for dead thread.